### PR TITLE
onnx: add relu/softmax/maxpool

### PR DESCRIPTION
**Pull request by Herald-of-Aurora**

### CODE REVIEW

File: pyzoo/zoo/pipeline/api/onnx/mapper/maxpool.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+class MaxPoolMapper(OperatorMapper):
+    def __init__(self, node, _params, _all_tensors):
+        super(MaxPoolMapper, self).__init__(node, _params, _all_tensors)
+
+    def format_params(self, params):
+        """
+        Convert ONNX params to Zoo format
+        :return: list of ndarray
+        """
+        return [np.expand_dims(params[0], 0), params[1]]
```
Comment (zhichao-li) [inline pyzoo/zoo/pipeline/api/onnx/mapper/maxpool.py]: remove this

File: pyzoo/zoo/pipeline/api/onnx/mapper/relu.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+class ReluMapper(OperatorMapper):
+    def __init__(self, node, _params, _all_tensors):
+        super(ReluMapper, self).__init__(node, _params, _all_tensors)
+
+    def create_operator(self):
+        assert len(self.inputs) == 1, "Conv accept single input only"
+        rank = len(self.inputs[0].get_input_shape())
+
+        if (rank == 2):  # NCHW
```
Comment (zhichao-li) [inline pyzoo/zoo/pipeline/api/onnx/mapper/relu.py]: remove this

File: pyzoo/zoo/pipeline/api/onnx/mapper/relu.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+            relu = zlayers.Activation("relu")
+            return relu
+
+        else:
```
Comment (zhichao-li) [inline pyzoo/zoo/pipeline/api/onnx/mapper/relu.py]: remove

File: pyzoo/zoo/pipeline/api/onnx/mapper/softmax.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+class SoftmaxMapper(OperatorMapper):
+    def __init__(self, node, _params, _all_tensors):
+        super(SoftmaxMapper, self).__init__(node, _params, _all_tensors)
+
+    def create_operator(self):
+        assert len(self.inputs) == 1, "Conv accept single input only"
+        rank = len(self.inputs[0].get_input_shape())
+
+        if (rank == 2):  # NCHW
```
Comment (zhichao-li) [inline pyzoo/zoo/pipeline/api/onnx/mapper/softmax.py]: remove

File: pyzoo/zoo/pipeline/api/onnx/mapper/maxpool.py [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+        if (rank == 4):  # NCHWinputs
+            pool_size = [int(i) for i in self.onnx_attr['kernel_shape']]
+            strides = [int(i) for i in self.onnx_attr['strides']]
+
+            border_mode, pads = OnnxHelper.get_padds(self.onnx_attr)
+            if border_mode is None:
+                border_mode = "valid"
+
+            maxpool = zlayers.MaxPooling2D(pool_size = pool_size,
```
Comment (zhichao-li) [inline pyzoo/zoo/pipeline/api/onnx/mapper/maxpool.py]: support `pads` ?

File: pyzoo/zoo/pipeline/api/onnx/mapper/operator_mapper.py [hunk 6]
```diff
@@ -51,10 +51,12 @@ def action(self):
         operator.set_name(self.node_name)
         if not isinstance(operator, zautograd.Variable):
             z_tensor = operator(self.inputs)
-            operator.set_weights(self.format_params(self.params))
+            if self.params is None:
```
Comment (zhichao-li) [inline pyzoo/zoo/pipeline/api/onnx/mapper/operator_mapper.py]: change to if `self.params` ?

File: pyzoo/zoo/pipeline/api/onnx/onnx_helper.py [hunk 7]
```diff
@@ -69,4 +69,6 @@ def get_padds(onnx_attr):
             assert pads4[0] == pads4[1]
             assert pads4[2] == pads4[3]
             pads = [pads4[0], pads4[1]]
```
Comment (zhichao-li) [inline pyzoo/zoo/pipeline/api/onnx/onnx_helper.py]:
```
elif onnx_attr['auto_pad'] == 'VALID':
                border_mode = 'valid'
else:
    exception

File: pyzoo/zoo/pipeline/api/onnx/onnx_helper.py [hunk 8]
```diff
@@ -62,11 +62,17 @@ def get_padds(onnx_attr):
                 border_mode = 'same'
             elif onnx_attr['auto_pad'] == 'SAME_LOWER':
```
Comment (zhichao-li) [inline pyzoo/zoo/pipeline/api/onnx/onnx_helper.py]: remove ` elif onnx_attr['auto_pad'] == 'SAME_LOWER':`

File: pyzoo/zoo/pipeline/api/onnx/onnx_helper.py [hunk 9]
```diff
@@ -62,11 +62,17 @@ def get_padds(onnx_attr):
...
                 raise NotImplementedError('same_lower auto_pad is not implemented')
+            elif onnx_attr['auto_pad'] == 'VALID':
+                border_mode = 'valid'
+            else:
+                raise NotImplementedError('unknown auto_pad mode')
```
Comment (zhichao-li) [inline pyzoo/zoo/pipeline/api/onnx/onnx_helper.py]: `s"Unknown auto_pad mode ${onnx_attr['auto_pad'] }"`

**Comment by Herald-of-Aurora** [comment]
Jenkins passed, id: 1015. (/view/ZOO-PR/job/ZOO-PR-Validation/1015/)

**Pull request merged**